scripts/build.py

```diff
@@ -46,12 +46,15 @@
 
 EXTENSIONS_DIRNAMES_TO_DIRPATHS = {
     'dev_dir': os.path.join('extensions', ''),
+    'dev_dir_for_js': os.path.join('compiled_js', 'extensions', ''),
     'staging_dir': os.path.join('backend_prod_files', 'extensions', ''),
     'out_dir': os.path.join('build', 'extensions', '')
 }
 TEMPLATES_DEV_DIR = os.path.join('templates', 'dev', 'head', '')
 TEMPLATES_CORE_DIRNAMES_TO_DIRPATHS = {
     'dev_dir': os.path.join('core', 'templates', 'dev', 'head', ''),
+    'dev_dir_for_js': os.path.join(
+        'compiled_js', 'core', 'templates', 'dev', 'head', ''),
```

Comment:
Suggest local_compiled_js/, and put under .gitignore.

Use dev dir for files in the main codebase; use staging dir for minified files; use out dir for final version.